appserver/extras/embedded/all/pom.xml

```diff
@@ -1382,6 +1382,12 @@
             <groupId>org.apache.santuario</groupId>
             <artifactId>xmlsec</artifactId>
         </dependency>
+        <!-- xmlsec depends on commons-codec -->
```

Comment:
Shouldn't it come from `xmlsec` without that declaration? Why do we need to have it explicitly?

Reply:
It is a workaround to avoid downloading all dependencies when they are repackaged inside. 
In the future we might consider also to avoid repackaging dependencies except glassfish.main, but everything has consequences.
Also Maven 4 will separate pom used for build and pom used for deployment. Originally we discussed to use some maven plugin for that with Maven 3 ... again ... has consequences.

Reply:
Yet one thing about Apache Commons - I always loved them, but it seems I need them less and less as I use newer JDK versions. We could consider if the dependency couldn't be removed from our dependencies.

Reply:
> Originally we discussed to use some maven plugin for that with Maven 3 ... again ... has consequences.

The maven flatten plug-in, right? I started to use it in several projects.